app/models/visitors_step.rb

```diff
@@ -56,7 +59,9 @@ def visitors_attributes=(params)
 
   def valid?(*)
     # This must be eager because we want to show errors on all objects.
-    visitors.inject([super]) { |a, e| a << e.valid? }.all?
+    visitors_valid = visitors.map(&:valid?).all?
```

Comment:
A comment saying that the order of validating visitors and step matters might help in the future. Also, do we have a test that catches if the order changes?